src/test/test-condition.c

```diff
@@ -323,6 +327,108 @@ static void test_condition_test_virtualization(void) {
         }
 }
 
+static void test_condition_test_uid(void) {
+        Condition *condition;
+        char* uid;
+        int r;
+
+        condition = condition_new(CONDITION_UID, "garbage oifdsjfoidsjoj", false, false);
+        assert_se(condition);
+        r = condition_test(condition);
+        log_info("ConditionUID=garbage → %i", r);
+        assert_se(r == -EINVAL);
+        condition_free(condition);
+
+        assert_se(0 < asprintf(&uid, "%u", UINT32_C(0xFFFF)));
```

Comment:
If you use UINT32_C() you probably should use PRIu32 to format the value, not `%u`...